app/uk/gov/hmrc/helptosave/models/HTSEvent.scala

```diff
@@ -18,85 +18,144 @@ package uk.gov.hmrc.helptosave.models
+    case None                               ⇒ Json.toJson(EligibilityResult(result.nino, result.eligible))
   }
 }
 
+case class EligibilityResult(nino:                String,
+                             eligible:            Boolean,
+                             ineligibleReason:    Option[String]  = None,
+                             isUCClaimant:        Boolean         = false,
+                             isWithinUCThreshold: Option[Boolean] = None)
+
+object EligibilityResult {
+
+  implicit val format: Format[EligibilityResult] = Json.format[EligibilityResult]
+}
+
 case class AccountCreated(userInfo: NSIPayload, source: String)(implicit hc: HeaderCarrier, appConfig: AppConfig) extends HTSEvent {
 
   private val createAccountURL = routes.HelpToSaveController.createAccount().url
 
-  val value: DataEvent = HTSEvent(
+  val value: ExtendedDataEvent = HTSEvent(
     appConfig.appName,
     "AccountCreated",
-    Map[String, String](
-      "forename" → userInfo.forename,
-      "surname" → userInfo.surname,
-      "dateOfBirth" → userInfo.dateOfBirth.toString,
-      "nino" → userInfo.nino,
-      "address1" → userInfo.contactDetails.address1,
-      "address2" → userInfo.contactDetails.address2,
-      "address3" → userInfo.contactDetails.address3.getOrElse(""),
-      "address4" → userInfo.contactDetails.address4.getOrElse(""),
-      "address5" → userInfo.contactDetails.address5.getOrElse(""),
-      "postcode" → userInfo.contactDetails.postcode,
-      "countryCode" → userInfo.contactDetails.countryCode.getOrElse(""),
-      "email" → userInfo.contactDetails.email.getOrElse(""),
-      "phoneNumber" → userInfo.contactDetails.phoneNumber.getOrElse(""),
-      "communicationPreference" → userInfo.contactDetails.communicationPreference,
-      "registrationChannel" → userInfo.registrationChannel,
-      "source" → source
+    Json.toJson(AllDetails(ExistingDetails(
+      userInfo.forename,
+      userInfo.surname,
+      userInfo.dateOfBirth.toString,
+      userInfo.nino,
+      userInfo.contactDetails.address1,
+      userInfo.contactDetails.address2,
+      userInfo.contactDetails.address3.getOrElse(""),
+      userInfo.contactDetails.address4.getOrElse(""),
+      userInfo.contactDetails.address5.getOrElse(""),
+      userInfo.contactDetails.postcode,
+      userInfo.contactDetails.countryCode.getOrElse(""),
+      userInfo.contactDetails.email.getOrElse(""),
+      userInfo.contactDetails.phoneNumber.getOrElse(""),
+      userInfo.contactDetails.communicationPreference,
+      userInfo.registrationChannel,
+      source
     ),
+                           Some(ManuallyEnteredDetails(
+        userInfo.nbaDetails.fold("")(_.accountName),
+        userInfo.nbaDetails.fold("")(_.accountNumber),
+        userInfo.nbaDetails.fold("")(_.sortCode),
+        userInfo.nbaDetails.fold("")(_.rollNumber.getOrElse(""))
```

Comment:
to avoid both the `fold` and `getOrElse` you could do `userInfo.nbaDetails.flatMap(_.rollNumber).getOrElse("")

Reply:
ok thanks